airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -55,21 +56,20 @@ def url_base(self) -> str:
         return f"https://{self.shop}.myshopify.com/admin/api/{self.api_version}/"
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        decoded_response = response.json()
-        if len(decoded_response.get(self.data_field)) < self.limit:
-            return None
+        # Getting next page link
+        next_page = response.links.get("next", None)
+        if next_page:
+            return dict(parse_qsl(urlparse(next_page.get("url")).query))
         else:
-            self.since_id = decoded_response.get(self.data_field)[-1]["id"]
-            return {"since_id": self.since_id}
+            return None
 
     def request_params(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
     ) -> MutableMapping[str, Any]:
-        params = {"limit": self.limit, "order": f"{self.primary_key} asc", "created_at_min": self.start_date}
         if next_page_token:
-            params.pop("created_at_min", None)
-            params.pop("order", None)
-            params.update(**next_page_token)
+            params = {"limit": self.limit, **next_page_token}
+        else:
+            params = {"limit": self.limit, "order": f"{self.primary_key} asc", "created_at": self.start_date}
```

Comment:
this should use `updated_at` rather than `created_at` to replicate records which received updates after creation?

Reply:
Reasonable. Corrected to `updated_at` in this place